docs/rfc/rfc-018-bls-agg-exploration.md

```diff
@@ -0,0 +1,400 @@
+One for the yays and one for the nays. Each BLS aggregated signature is 48
+bytes, per the [IETF standard of BLS signatures][bls-ietf-ecdsa-compare].
+Over the lifetime of the same cosmos hub chain, that would amount to about 1
+GB, a savings of 112 GB. While that is a large factor of reduction it's worth
+bearing in mind that, at [GCP's cost][gcp-storage-pricing] of $.026 USD per GB,
+that is a total savings of around $2.50 per month.
+
+
+
+
+#### Reduce Signature Verification Creation and Verification Time
+
+From the [IETF draft standard on BLS Signatures][bls-ietf], BLS signatures can be
+created in 370 microseconds and verified in 2700 microseconds. Our current
+[Ed25519 implementation][voi-ed25519-perf] lists a 27.5 microsecond signature creation time
+and 5.19 milliseconds to perform batch verification on 128 signatures, which is
+slightly fewer than the 175 in the hub. blst, a popular implementation of BLS
+signature aggregation was benchmarked to perform verification on 100 signatures in
+1.5 milliseconds [when run locally][blst-verify-bench] on an 8 thread machine and
+pre-aggregated public keys.
+
+#### Reduce Light-Client Verification Time
+
+The light client aims to be a faster and lighter-weight way to verify that a 
+block was voted on by a Tendermint network. The light client fetches
+Tendermint block headers and commit signatures, performing public key
+verification to ensure that the associated validator set signed the block.
+Reducing the size of the commit signature would allow the light client to fetch
+block data more quickly.
+
+Additionally, the faster signature verification times of BLS signatures mean
+that light client verification would proceed more quickly.
+
+#### Reduce Gossip Bandwidth
+
+##### Vote Gossip
+
+It is possible to aggregate signatures during voting and not need to gossip all 
+*n* validator signatures to all other validators. Theoretically, subsets of
+the signatures could be aggregated during consensus to produce vote messages
+carrying aggregated signatures. Implementations of this would certainly
+increase the complexity of the gossip layer but may possibly reduce the total
+number of signatures required to be verified by each validator.
+
+##### Block Gossip
+
+A reduction in the block size as a result of signature aggregation would
+naturally lead to a reduction in the bandwidth required to gossip a block.
+Each validator would only send and receive the smaller aggregated signatures
+instead of the full list of multi-signatures as we have them now.
+
+### What are the drawbacks to aggregated signatures?
+
+#### Heterogeneous key types cannot be aggregated
+
+Only one type of signature can be aggregated and our legacy signing schemes
+cannot be aggregated. In practice, this means that aggregated signatures can
+be created over the set of all validators that use BLS signatures and validators
+with alternative key types, such as Ed25519 must be included separately in
+blocks and votes.
```

Comment:
```suggestion
Aggregation requires a specific signature algorithm, and our legacy signing schemes
cannot be aggregated. In practice, this means that aggregated signatures could
be created for a subset of validators using BLS signatures, and validators
with other key types (such as Ed25519) would still have to be be separately
propagated in blocks and votes.
```

Related to this: There is a hairy migration problem even for validators that WANT to use the BLS scheme, to get from their current keys to the new thing. 

(Of course, they could hard fork, but are the savings from this likely to justify the network-wide costs of a hard fork?)